src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -1,128 +1,17 @@
 import { EIP1193 } from '@web3-react/eip1193'
-import { EMPTY } from '@web3-react/empty'
 import { MetaMask } from '@web3-react/metamask'
+import { Network } from '@web3-react/network'
 import { Connector, Provider as Eip1193Provider, Web3ReactStore } from '@web3-react/types'
 import { Url } from '@web3-react/url'
 import { WalletConnect } from '@web3-react/walletconnect'
 import { Buffer } from 'buffer'
-import { createContext, PropsWithChildren, useContext, useMemo } from 'react'
+import { SupportedChainId } from 'constants/chains'
+import { PropsWithChildren, useMemo } from 'react'
 import { useCallback } from 'react'
-import JsonRpcConnector from 'utils/JsonRpcConnector'
-
-export type Web3ContextType = {
-  connector: Connector
-  library?: (JsonRpcProvider & { provider?: ExternalProvider }) | Web3Provider
-  chainId?: ReturnType<Web3ReactHooks['useChainId']>
-  accounts?: ReturnType<Web3ReactHooks['useAccounts']>
-  account?: ReturnType<Web3ReactHooks['useAccount']>
-  active?: ReturnType<Web3ReactHooks['useIsActive']>
-  activating?: ReturnType<Web3ReactHooks['useIsActivating']>
-}
-
-const [EMPTY_CONNECTOR, EMPTY_HOOKS] = initializeConnector<Connector>(() => EMPTY)
-const EMPTY_STATE = { connector: EMPTY_CONNECTOR, hooks: EMPTY_HOOKS }
-const EMPTY_WEB3: Web3ContextType = { connector: EMPTY }
-const EMPTY_CONTEXT = { web3: EMPTY_WEB3, updateWeb3: (updateContext: Web3ContextType) => console.log(updateContext) }
-const Web3Context = createContext(EMPTY_CONTEXT)
-
-export default function useActiveWeb3React() {
-  const { web3 } = useContext(Web3Context)
-  return web3
-}
-
-function useUpdateActiveWeb3ReactCallback() {
-  const { updateWeb3 } = useContext(Web3Context)
-  return updateWeb3
-}
-
-function getNetwork(jsonRpcEndpoint?: string | JsonRpcProvider) {
-  if (jsonRpcEndpoint) {
-    let connector, hooks
-    if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
-      ;[connector, hooks] = initializeConnector((actions) => new JsonRpcConnector(actions, jsonRpcEndpoint))
-    } else {
-      ;[connector, hooks] = initializeConnector((actions) => new Url({ actions, url: jsonRpcEndpoint }))
-    }
-    connector.activate()
-    return { connector, hooks }
-  }
-  return EMPTY_STATE
-}
-
-function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
-  if (provider) {
-    let connector, hooks
-    if (JsonRpcProvider.isProvider(provider)) {
-      ;[connector, hooks] = initializeConnector((actions) => new JsonRpcConnector(actions, provider))
-    } else if (JsonRpcProvider.isProvider((provider as any).provider)) {
-      throw new Error('Eip1193Bridge is experimental: pass your ethers Provider directly')
-    } else {
-      ;[connector, hooks] = initializeConnector((actions) => new EIP1193({ actions, provider }))
-    }
-    connector.activate()
-    return { connector, hooks }
-  }
-  return EMPTY_STATE
-}
 
 export let connections: [Connector, Web3ReactHooks][] = []
```

Comment:
you should have this be a function called `getConnections` instead of having it live as mutable at the top level